sympy/printing/printer.py

```diff
@@ -55,135 +61,131 @@
-            user defined class subclassed from the SymPy Basic class.
-            \"\"\"
+    class ModOpWrong(Mod):
+        def _latex(self, printer=None):
+            a, b = [str(i) for i in self.args]
+            return r"\\operatorname{Mod}{\\left( %s,%s \\right)}" % (a,b)
 
-            def _print_Derivative(self, expr):
-                \"\"\"
-                Custom printing of the SymPy Derivative class.
 
-                Instead of:
+    x = Symbol('x')
+    m = Symbol('m')
 
-                D(x(t), t) or D(x(t), t, t)
+    print_latex(ModOp(x, m))
+    print_latex(Mod(x, m))
 
-                We will print:
+    # Nested modulo.
+    print_latex(ModOp(ModOp(x, m), Integer(7)))
+    print_latex(ModOpWrong(ModOpWrong(x, m), Integer(7)))
 
-                x'     or     x''
+The output of the code above is::
 
-                In this example, expr.args == (x(t), t), and expr.args[0] == x(t), and
-                expr.args[0].func == x
-                \"\"\"
-                return str(expr.args[0].func) + "'"*len(expr.args[1:])
+    \\operatorname{Mod}{\\left( x,m \\right)}
+    x\\bmod{m}
+    \\operatorname{Mod}{\\left( \\operatorname{Mod}{\\left( x,m \\right)},7 \\right)}
+    \\operatorname{Mod}{\\left( ModOpWrong(x, m),7 \\right)}
+"""
 
-            def _print_MyClass(self, expr):
-                \"\"\"
-                Print the characters of MyClass.s alternatively lower case and upper
-                case
-                \"\"\"
-                s = ""
-                i = 0
-                for char in expr.s:
-                    if i % 2 == 0:
-                        s += char.lower()
-                    else:
-                        s += char.upper()
-                    i += 1
-                return s
+from __future__ import print_function, division
 
-        # Override the __str__ method of to use CustromStrPrinter
-        Basic.__str__ = lambda self: CustomStrPrinter().doprint(self)
-        # Demonstration of CustomStrPrinter:
-        t = Symbol('t')
-        x = Function('x')(t)
-        dxdt = x.diff(t)            # dxdt is a Derivative instance
-        d2xdt2 = dxdt.diff(t)       # dxdt2 is a Derivative instance
-        ex = MyClass('I like both lowercase and upper case')
+from sympy import Basic, Add
 
-        print dxdt
-        print d2xdt2
-        print ex
+from sympy.core.core import BasicMeta
+
+from functools import cmp_to_key
 
-    The output of the above code is::
 
-        x'
-        x''
-        i lIkE BoTh lOwErCaSe aNd uPpEr cAsE
+class Printer(object):
+    """ Generic printer
 
-    By overriding Basic.__str__, we can customize the printing of anything that
-    is subclassed from Basic.
+    Its job is to provide infrastructure for implementing new printers easily.
 
+    If you want to define your custom Printer or your custom printing method
+    for your custom class then see the example above: printer_example_ .
```

Comment:
an underscore is need here?  `printer_example_` shouldn't it be `_printer_example`?

Reply:
Well, that's the way internal links in sphinx work. When I changed it to `_printer_example` then the link stopped working.